exercises/ledger/canonical-data.json

```diff
@@ -0,0 +1,240 @@
+            "description": "Buy present",
+            "amountInCents": -1000
+          }
+        ]
+      },
+      "expected": [
+        "Date       | Description               | Change       ",
+        "01/01/2015 | Buy present               |      ($10.00)",
+        "01/02/2015 | Get present               |       $10.00 "
+      ]
+    },
+    {
+      "uuid": "29dd3659-6c2d-4380-94a8-6d96086e28e1",
+      "description": "final order tie breaker is change",
+      "property": "formatEntries",
+      "input": {
+        "currency": "USD",
+        "locale": "en_US",
+        "entries": [
+          {
+            "date": "2015-01-01",
+            "description": "Something",
+            "amountInCents": 0
+          },
+          {
+            "date": "2015-01-01",
+            "description": "Something",
+            "amountInCents": -1
+          },
+          {
+            "date": "2015-01-01",
+            "description": "Something",
+            "amountInCents": 1
+          }
+        ]
+      },
+      "expected": [
+        "Date       | Description               | Change       ",
+        "01/01/2015 | Something                 |       ($0.01)",
+        "01/01/2015 | Something                 |        $0.00 ",
+        "01/01/2015 | Something                 |        $0.01 "
+      ]
+    },
+    {
+      "uuid": "9b9712a6-f779-4f5c-a759-af65615fcbb9",
+      "description": "overlong description is truncated",
+      "property": "formatEntries",
+      "input": {
+        "currency": "USD",
+        "locale": "en_US",
+        "entries": [
+          {
+            "date": "2015-01-01",
+            "description": "Freude schoner Gotterfunken",
+            "amountInCents": -123456
+          }
+        ]
+      },
+      "expected": [
+        "Date       | Description               | Change       ",
+        "01/01/2015 | Freude schoner Gotterf... |   ($1,234.56)"
+      ]
+    },
+    {
+      "uuid": "67318aad-af53-4f3d-aa19-1293b4d4c924",
+      "description": "euros",
+      "property": "formatEntries",
+      "input": {
+        "currency": "EUR",
+        "locale": "en_US",
+        "entries": [
+          {
+            "date": "2015-01-01",
+            "description": "Buy present",
+            "amountInCents": -1000
+          }
+        ]
+      },
+      "expected": [
+        "Date       | Description               | Change       ",
+        "01/01/2015 | Buy present               |      (€10.00)"
+      ]
+    },
+    {
+      "uuid": "bdc499b6-51f5-4117-95f2-43cb6737208e",
+      "description": "dutch locale",
+      "property": "formatEntries",
+      "input": {
+        "currency": "USD",
+        "locale": "nl_NL",
+        "entries": [
+          {
+            "date": "2015-03-12",
+            "description": "Buy present",
+            "amountInCents": 123456
+          }
+        ]
+      },
+      "expected": [
+        "Datum      | Omschrijving              | Verandering  ",
```

Comment:
Verandering has the advantage of being consistent with all the existing Python solutions (and Go, C#, F#, Java!) :) Unless it's _wrong_, I'd prefer to avoid breaking existing tests.

Reply:
Yeah, I also feel it is a bit Google Translate-y. 

> Unless it's wrong, I'd prefer to avoid breaking existing tests.

Well, technically you wouldn't, but these are new test cases. Tracks could then opt into changing them or not.

I don't massively care, but `Bedrag`, `Mutatie` and `Wijziging` all sound good.

Reply:
I expect to be able to auto-generate unit tests which use the canonical data with minimal to little data changes. I expect those generated tests to continue passing for solutions which currently pass.

Ideally, I expect the generated unit tests to be pretty close to the existing unit tests. (I wrote this PR alongside the Python test generator and my expectations are based on the work I've already done and the Python generated test I'm already producing; I expect this to track fairly closely with other tracks which Python used as a test data source, or which used Python as a test data source.)

The headers are a hard coded requirement in the unit tests and are a hard coded string in solutions. Changing the headers will break [existing solutions](https://exercism.org/tracks/python/exercises/ledger/solutions).

Reply:
> Changing the headers will break [existing solutions](https://exercism.org/tracks/python/exercises/ledger/solutions).

Right, of course 🤦

Keeping the old headers makes sense.